patchwork-level-generators/src/main/java/net/patchworkmc/mixin/levelgenerators/MixinBiomeLayers.java

```diff
@@ -0,0 +1,114 @@
+			continentLayer = AddClimateLayers.AddSpecialBiomesLayer.INSTANCE.create(contextProvider.apply(3L), continentLayer);
+			continentLayer = ScaleLayer.NORMAL.create(contextProvider.apply(2002L), continentLayer);
+			continentLayer = ScaleLayer.NORMAL.create(contextProvider.apply(2003L), continentLayer);
+			continentLayer = IncreaseEdgeCurvatureLayer.INSTANCE.create(contextProvider.apply(4L), continentLayer);
+			continentLayer = AddMushroomIslandLayer.INSTANCE.create(contextProvider.apply(5L), continentLayer);
+			continentLayer = AddDeepOceanLayer.INSTANCE.create(contextProvider.apply(4L), continentLayer);
+			continentLayer = stack(1000L, ScaleLayer.NORMAL, continentLayer, 0, contextProvider);
+			int biomeSize = 4;
+			int riverSize = biomeSize;
+
+			if (settings != null) {
+				biomeSize = settings.getBiomeSize();
+				riverSize = settings.getRiverSize();
+			}
+
+			if (generatorType == LevelGeneratorType.LARGE_BIOMES) {
+				biomeSize = 6;
+			}
+
+			LayerFactory<T> riverLayer = stack(1000L, ScaleLayer.NORMAL, continentLayer, 0, contextProvider);
+			riverLayer = SimpleLandNoiseLayer.INSTANCE.create(contextProvider.apply(100L), riverLayer);
+			LayerFactory<T> noiseLayer = ((IForgeWorldType) generatorType).getBiomeLayer(continentLayer, settings, contextProvider);
+			LayerFactory<T> layerFactory5 = stack(1000L, ScaleLayer.NORMAL, riverLayer, 2, contextProvider);
+			noiseLayer = AddHillsLayer.INSTANCE.create(contextProvider.apply(1000L), noiseLayer, layerFactory5);
+			riverLayer = stack(1000L, ScaleLayer.NORMAL, riverLayer, 2, contextProvider);
+			riverLayer = stack(1000L, ScaleLayer.NORMAL, riverLayer, riverSize, contextProvider);
+			riverLayer = NoiseToRiverLayer.INSTANCE.create(contextProvider.apply(1L), riverLayer);
+			riverLayer = SmoothenShorelineLayer.INSTANCE.create(contextProvider.apply(1000L), riverLayer);
+			noiseLayer = AddSunflowerPlainsLayer.INSTANCE.create(contextProvider.apply(1001L), noiseLayer);
+
+			for (int k = 0; k < biomeSize; ++k) {
+				noiseLayer = ScaleLayer.NORMAL.create(contextProvider.apply((long) (1000 + k)), noiseLayer);
+
+				if (k == 0) {
+					noiseLayer = IncreaseEdgeCurvatureLayer.INSTANCE.create(contextProvider.apply(3L), noiseLayer);
+				}
+
+				if (k == 1 || biomeSize == 1) {
+					noiseLayer = AddEdgeBiomesLayer.INSTANCE.create(contextProvider.apply(1000L), noiseLayer);
+				}
+			}
+
+			noiseLayer = SmoothenShorelineLayer.INSTANCE.create(contextProvider.apply(1000L), noiseLayer);
+			noiseLayer = AddRiversLayer.INSTANCE.create(contextProvider.apply(100L), noiseLayer, riverLayer);
+			noiseLayer = ApplyOceanTemperatureLayer.INSTANCE.create(contextProvider.apply(100L), noiseLayer, layerFactory2);
+			LayerFactory<T> biomeLayer = CellScaleLayer.INSTANCE.create(contextProvider.apply(10L), noiseLayer);
+			info.setReturnValue(ImmutableList.of(noiseLayer, biomeLayer, noiseLayer));
```

Comment:
There's probably a better way to do this than copying all of the decompiled ARR mojang code...

Reply:
I don't think so. I have to cut out multiple lines and replace them with a single line

Reply:
ok I made this four redirects lol